packages/babylon/src/plugins/typescript.js

```diff
@@ -1917,7 +1925,9 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     isClassProperty(): boolean {
-      return this.match(tt.colon) || super.isClassProperty();
+      return (
+        this.match(tt.bang) || this.match(tt.colon) || super.isClassProperty()
```

Comment:
Weird, how does `?` get parsed from here?

Reply:
I think that it is done in parsePostMemberNameModifiers